### PR TITLE
feat(challenger): Challenger Driver

### PR DESCRIPTION
**Pull request by jackchuma**
### What changed? Why?

Implements the challenger driver — the core orchestration loop that ties together all challenger components into a fully functional service.

**Driver (`driver.rs`)**
- New `Driver` struct that runs a polling loop: scan for invalid dispute games, validate intermediate output roots, request ZK proofs, and submit nullification transactions
- Manages in-flight proof sessions via `PendingProof` with a two-phase lifecycle (`AwaitingProof` → `ReadyToSubmit`)
- Handles retries: failed ZK proofs are re-requested next tick; failed nullification txs are retried without re-proving
- Checks game status and `zk_prover` before each submission to avoid wasted work on already-resolved or already-challenged games
- Supports graceful shutdown via `CancellationToken`

**Service wiring (`service.rs`)**
- `ChallengerService::run()` is now fully wired: creates L1/L2 providers, contract clients (DisputeGameFactory, AggregateVerifier), ZK proof client, scanner, validator, and driver
- Installs a signal handler via `RuntimeManager` for graceful shutdown with `CancellationToken`
- Health server readiness flag is set to `true` once the driver is assembled and ready to run

**Scanner enhancement (`scanner.rs`)**
- `CandidateGame` now includes `intermediate_block_interval` so the driver can compute the correct block range when requesting ZK proofs
- `GameScanner` caches `game_type → intermediate_block_interval` mappings to avoid repeated RPC calls, resolved via `game_impls()` + `read_intermediate_block_interval()`

**Contract client (`aggregate_verifier.rs`)**
- Added `intermediate_output_roots()` to the `AggregateVerifierClient` trait and its concrete implementation, which reads and parses the concatenated 32-byte hashes from the game contract

**Health server (`health.rs`)**
- Now accepts a `CancellationToken` for graceful shutdown instead of relying on task abort

**CLI simplification**
- Removed `--rollup-rpc` and `--anchor-state-registry-addr` CLI args and config fields (no longer needed — the verifier address is resolved from the factory, and rollup RPC is unused)

**Test utilities**
- Extracted shared test helpers (`MockTxManager`, `receipt_with_status`, `TrieAccount`, `build_test_header_and_account`) into `test_utils` for reuse across modules
- Added `MockZkProofProvider` for driver tests
- Comprehensive driver test coverage: empty scans, valid games, block-not-available errors, proof success/failure, nullification retries, game resolution checks, cancellation

### Notes to reviewers

The driver is generic over `L2Provider`, `ZkProofProvider`, and `TxManager` traits, making it fully testable with mocks. The `poll_or_submit` method is the key state machine — it advances a pending proof through its lifecycle and handles all edge cases (game resolved, already challenged, proof failed, tx failed).

### How has it been tested?

- 10 unit tests covering the driver's scan-validate-prove-submit pipeline
- Tests for edge cases: empty scans, valid games skipped, block-not-available errors, proof success/failure, nullification tx failure with retry, resolved/challenged game detection, cancellation
- Existing scanner, validator, submitter, and health server tests updated and passing

### Change management ([definitions](http://go/change-management))

type=routine
risk=low
impact=sev5

### Backwards Compatibility ([learn more](http://go/backwards-compatibility))

backwards_compatible=false

Breaking: removes `--rollup-rpc` and `--anchor-state-registry-addr` CLI flags. These are no longer needed as the verifier address is resolved from the DisputeGameFactory.

automerge=false

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>✅ Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
✅
<code>1/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: crates/proof/challenge/src/driver.rs
Comment (refcell) [inline crates/proof/challenge/src/driver.rs:1]: Can we split this file up a bit? Isolated types could go in their own modules for example

File: crates/proof/challenge/src/driver.rs [hunk 1]
```diff
@@ -0,0 +1,941 @@
...
+                // Leave entry as ReadyToSubmit for retry.
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (refcell) [inline crates/proof/challenge/src/driver.rs]: Maybe some of these can be moved into integration tests in `crates/proof/challenge/tests/` to make this file a bit smaller?

File: crates/proof/challenge/src/driver.rs [hunk 2]
```diff
@@ -0,0 +1,941 @@
...
+                    other => {
+                        debug!(
+                            game = %game_address,
+                            session_id = %session_id,
+                            status = ?other,
+                            "proof not ready, will retry next tick"
+                        );
+                        return Ok(());
+                    }
+                }
```
Comment (refcell) [inline crates/proof/challenge/src/driver.rs]: This could be a candidate to refactor onto a new `PendingProofs` type to make this file a bit slimmer.

e.g.
```rust
impl PendingProofs {
  /// Updates pending proofs from a [`ProofJobStatus`].
  pub fn update_from(&mut self, status: ProofJobStatus) -> Option<Bytes> {
    // match on status, return bytes or none if failed/other
  }
}
```

**Review by refcell — APPROVED** [review]
This LGTM
